tests/Model/LazyLoading/AbstractLazyLoadingTest.php

```diff
@@ -117,14 +117,17 @@ protected function loadSingleRelation(): RelationTest
         return $listing->load()[0];
     }
 
-    protected function checkSerialization(LazyLoading $object, string $messagePrefix, bool $contentShouldBeIncluded = false)
+    protected function checkSerialization(LazyLoading $object, string $messagePrefix, bool $contentShouldBeIncluded = false): void
     {
         $serializedString = serialize($object);
         $this->checkSerializedStringForNeedle($serializedString, ['lazyLoadedFields', 'lazyKeys', 'loadedLazyKeys'], false, $messagePrefix);
         $this->checkSerializedStringForNeedle($serializedString, 'someAttribute";s:14:"Some content', $contentShouldBeIncluded, $messagePrefix);
     }
 
-    protected function checkSerializedStringForNeedle(string $string, $needle, bool $expected, string $messagePrefix = null)
+    /**
+     * @param string[]|string $needle
```

Comment:
Why adding the doc type here?

Reply:
It describes, that the needle array should be a list of strings. It is a best practice to doc the array values.

Reply:
Ahh okay, it just looked a little strange to me, if there are no doctypes at all but for the needle.